### PR TITLE
Add support for LibXSMM backend

### PR DESCRIPTION
**Pull request by krenzland**
Replaces the old (deprecated) LibXSMM backend with new JIT based backend.

Advantage: Runs on more machines, has more features (e.g. B can be transposed).

TODO: Need to fix CI, @ravil-mobile some input might be helpful here.
I will provide performance benchmarks and will run a small convergence test to ensure that it does not lead to regressions.

### CODE REVIEW

**Comment by krenzland** [comment]
@ravil-mobile Can you please install the standard interface of libxsmm in the Docker image used for this CI? I do not have access to it. Should work identically to installing the generator - just "make install" instead of "make generator && cp ..."

File: tests/code-gen/cmake/FindLIBXSMM.cmake [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+if (NOT(
+        LIBXSMM_INCLUDE_DIRS STREQUAL "LIBXSMM_INCLUDE_DIRS-NOTFOUND" OR
+        LIBXSMM_LIBARIES STREQUAL "LIBXSMM_LIBRARIES-NOTFOUND"))
+    add_library(LIBXSMM::LIBXSMM INTERFACE IMPORTED)
+    set_target_properties(LIBXSMM::LIBXSMM PROPERTIES
+            INTERFACE_LINK_LIBRARIES "${LIBXSMM_LIBRARIES}"
+            INTERFACE_INCLUDE_DIRECTORIES "${LIBXSMM_INCLUDE_DIRS}")
+else()
+    message("WTF?")
```
Comment (ravil-mobile) [inline tests/code-gen/cmake/FindLIBXSMM.cmake:20]: don't forget to delete

**Review by ravil-mobile — APPROVED** [review]

File: yateto/gemm_configuration.py [hunk 2]
```diff
@@ -240,18 +286,19 @@ class DefaultGeneratorCollection(GeneratorCollection):
...
-      'hsw' : [libxsmm, mkl, blis, eigen],
-      'rome' : [libxsmm, blis, eigen],
-      'knl' : [libxsmm, pspamm, mkl, blis, eigen],
-      'skx' : [libxsmm, pspamm, mkl, blis, eigen],
+      'snb' : [libxsmm_jit, libxsmm, mkl, blis, eigen],
+      'hsw' : [libxsmm_jit, libxsmm, mkl, blis, eigen],
+      'rome' : [libxsmm_jit, libxsmm, blis, eigen],
+      'knl' : [libxsmm_jit, libxsmm, pspamm, mkl, blis, eigen],
+      'skx' : [libxsmm_jit, libxsmm, pspamm, mkl, blis, eigen],
       'thunderx2t99' : [pspamm, openblas, blis, eigen],
```
Comment (ravil-mobile) [inline yateto/gemm_configuration.py:302]: btw, do you want to add `libxsmm_jit` here?

Comment (krenzland) [reply yateto/gemm_configuration.py:302]: Yes, after testing :)

**Pull request merged**